src/data_store/data_store_conduit.cpp

```diff
@@ -296,17 +301,21 @@ void data_store_conduit::set_conduit_node(int data_id, conduit::Node &node, bool
   if (m_is_local_cache && m_preload) {
     LBANN_ERROR("you called data_store_conduit::set_conduit_node, but you're running in local cache mode with preloading; something is broken; please contact Dave Hysom");
   }
-  //m_mutex.lock();
+
+  m_mutex.lock();
   if (already_have == false && m_data.find(data_id) != m_data.end()) {
+    m_mutex.unlock();
     LBANN_ERROR("duplicate data_id: ", data_id, " in data_store_conduit::set_conduit_node; role: ", m_reader->get_role());
   }
-
+  m_mutex.unlock();
```

Comment:
The methods of `std::mutex` should in general not be used directly; it is meant to be used in `std::unique_lock` or `std::lock_guard` (and so on), to provide exception-safe locking with RAII semantics.